### PR TITLE
Vicodyn tuning

### PR DESCRIPTION
**Pull request by antmat**

### CODE REVIEW

File: vicodyn/src/vicodyn/queue/invocation.cpp [hunk 1]
```diff
@@ -12,6 +12,28 @@ namespace cocaine {
...
+auto invocation_t::absorb(invocation_t&& other) -> void {
+    m_session.apply([&](std::shared_ptr<session_t>& session) {
+        if(!session) {
+            for(auto& op: other.m_operations) {
+                m_operations.push_back(std::move(op));
+            }
+        } else {
+            for (auto it = other.m_operations.begin(); it != other.m_operations.end();) {
+                try {
+                    execute(session, *it);
```
Comment (3Hren) [inline vicodyn/src/vicodyn/queue/invocation.cpp:24]: Infinite loop?

File: vicodyn/include/cocaine/api/peer/pool.hpp [hunk 2]
```diff
@@ -20,7 +20,7 @@ class pool_t {
     virtual
     auto invoke(const io::aux::decoded_message_t& incoming_message,
                 const io::graph_node_t& protocol,
-                io::upstream_ptr_t downstream) -> std::shared_ptr<vicodyn::queue::send_t> = 0;
+                io::upstream_ptr_t downstream) -> std::shared_ptr<cocaine::vicodyn::queue::send_t> = 0;
```
Comment (3Hren) [inline vicodyn/include/cocaine/api/peer/pool.hpp:23]: Seems unnecessary.

**Review by 3Hren — CHANGES_REQUESTED** [review]

File: vicodyn/src/vicodyn/queue/invocation.cpp [hunk 3]
```diff
@@ -12,6 +12,29 @@ namespace cocaine {
...
+                    it = other.m_operations.erase(it);
+                } catch (...) {
+                    it = other.m_operations.erase(it);
+                    session = nullptr;
+                    for(auto& op: other.m_operations) {
```
Comment (3Hren) [inline vicodyn/src/vicodyn/queue/invocation.cpp:29]: Error here.

**Review by 3Hren — CHANGES_REQUESTED** [review]

File: vicodyn/src/vicodyn/queue/invocation.cpp (hunk 1 above)
Comment (antmat) [reply vicodyn/src/vicodyn/queue/invocation.cpp:24]: Cupertino, CA 95014

**Pull request merged**